pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java

```diff
@@ -44,62 +51,65 @@
 @Slf4j
 public class NamespaceStatsAggregator {
 
-    private static FastThreadLocal<AggregatedNamespaceStats> localNamespaceStats =
-            new FastThreadLocal<AggregatedNamespaceStats>() {
+    private static final FastThreadLocal<AggregatedNamespaceStats> localNamespaceStats =
+            new FastThreadLocal<>() {
                 @Override
-                protected AggregatedNamespaceStats initialValue() throws Exception {
+                protected AggregatedNamespaceStats initialValue() {
                     return new AggregatedNamespaceStats();
                 }
             };
 
-    private static FastThreadLocal<TopicStats> localTopicStats = new FastThreadLocal<TopicStats>() {
+    private static final FastThreadLocal<TopicStats> localTopicStats = new FastThreadLocal<>() {
         @Override
-        protected TopicStats initialValue() throws Exception {
+        protected TopicStats initialValue() {
             return new TopicStats();
         }
     };
 
     public static void generate(PulsarService pulsar, boolean includeTopicMetrics, boolean includeConsumerMetrics,
-           boolean includeProducerMetrics, boolean splitTopicAndPartitionIndexLabel, SimpleTextOutputStream stream) {
+                                boolean includeProducerMetrics, boolean splitTopicAndPartitionIndexLabel,
+                                SimpleTextOutputStream stream) {
         String cluster = pulsar.getConfiguration().getClusterName();
         AggregatedNamespaceStats namespaceStats = localNamespaceStats.get();
-        TopicStats.resetTypes();
         TopicStats topicStats = localTopicStats.get();
-
-        printDefaultBrokerStats(stream, cluster);
-
         Optional<CompactorMXBean> compactorMXBean = getCompactorMXBean(pulsar);
         LongAdder topicsCount = new LongAdder();
+        Map<String, ByteBuf> allMetrics = new HashMap<>();
```

Comment:
Not sure I understand the motivation here of creating a SimpleTextOutputStream and a ByteBuf for each metric. Using SimpleTextOutputStream per metric was my original PR.

Reply:
The idea is to have a separate `ByteBuf` per metric, right? 
`SimpleTextOutputStream` is just a wrapper around an existing `ByteBuf`, which gives serialization functionality of writing int, long, ByteBuf, etc into a ByteBuf.
So instead of copy-pasting the functionality from `SimpleTextOutputStream` into `NamespaceStatsAggregator`, and using the copied functionality to write into the metric `ByteBuf`, you wrap that `ByteBuf` with `SimpleTextOutputStream` to gain that functionality.

So the change is: `Map<String, ByteBuf>` to `Map<String, SimpleTextOutputStream>`

Reply:
We would however still need to have some access to the `ByteBuf`'s to be able to release them?